docs/use-cases/microfrontends.md

```diff
@@ -0,0 +1,308 @@
+
+```js
+// featurevisor.config.js
+module.exports = {
+  environments: [
+    "staging",
+    "production",
+  ],
+
+  tags: [
+    'products',
+    'signup',
+    'signin',
+    'checkout',
+    'account'
+  ]
+};
+```
+
+Once this configuration is in place, we can [build](/docs/building-datafiles) our datafiles:
+
+```
+$ featurevisor build
+```
+
+And it will output the following datafiles in the `dist` directory:
+
+```
+$ tree dist
+dist
+├── production
+│   ├── datafile-tag-products.json
+│   ├── datafile-tag-signup.json
+│   ├── datafile-tag-signin.json
+│   ├── datafile-tag-checkout.json
+│   └── datafile-tag-account.json
+└── staging
+│   ├── datafile-tag-products.json
+│   ├── datafile-tag-signup.json
+│   ├── datafile-tag-signin.json
+│   ├── datafile-tag-checkout.json
+│   └── datafile-tag-account.json
+
+2 directories, 10 files
+```
+
+## Attributes
+
+We will set up some Featurevisor attributes first, that we will use in various stages of this guide later.
+
+### `userId`
+
+```yml
+# attributes/userId.yml
+description: Unique identifier of the logged in User
+type: string
+capture: true
+```
+
+### `deviceId`
+
+This ID can be generated at the client-side level when user first visits your app, and stored in a cookie or localStorage for e.g. if in a browser environment.
+
+```yml
+# attributes/deviceId.yml
+description: Unique identifier of the device
+type: string
+capture: true
+```
+
+## Feature
+
+For this example, we will create a new feature flag that's responsible for toggling a marketing banner that's shown at the bottom of a page. We can call it `showMarketingBanner`.
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - <your-tag-here> # we will discuss this in next section below
+
+defaultVariation: false
+
+bucketBy: deviceId
+
+variations:
+  - value: true
+    weight: 100
+
+  - value: false
+    weight: 0
+
+environments:
+  staging:
+    rules:
+      - key: "1"
+        segments: "*"
+        percentage: 100
+
+  production:
+    rules:
+      - key: "1"
+        segments: "*"
+        percentage: 100
+```
+
+## Tagging features
+
+When we create or update any feature in your Featurevisor project, we can tag it with the microfrontend it belongs to:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products # tagging with `products` microfrontend
+
+# ...
+```
+
+It is possible some features may overlap across microfrontends. For example, we may want to show a marketing banner to users in both the `products` and `checkout` microfrontends. In that case, we can tag the feature with both microfrontends:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products
+  - checkout
+
+# ...
+```
+
+## Anonymous vs Authenticated
```

Comment:
Playing advocate's devil here. Is this really about Microfrontend? I've got a feeling you'd have the same issue in a simple SPA app.

Reply:
you are right, this applies to any application (either monolithic or microfrontends) where we deal with users before and after they log in.

I don't have this use case documented anywhere else in the site that I can link to, and wanted to squeeze it in here for the use case of microfrontends (or parts of application) that can potentially be exposed to both anonymous and authenticated users.

Reply:
I hade a added a note under this heading's section at the bottom, mentioning the concern applies to both monoliths and microfrontends.